packages/expo-router/src/hooks.ts

```diff
@@ -92,7 +92,13 @@ export function useGlobalSearchParams<
 export function useLocalSearchParams<
   TParams extends SearchParams = SearchParams,
 >(): Partial<TParams> {
-  const params = React.useContext(NavigationRouteContext)?.params ?? {};
+  const routeInfo = React.useContext(NavigationRouteContext);
+  const params = routeInfo?.params ?? {};
+
+  if (routeInfo?.name === '+not-found') {
+    delete params['not-found'];
```

Comment:
Presumably users will want to be able to access this parameter, it just shouldn't show in the URL bar as a standalone parameter.